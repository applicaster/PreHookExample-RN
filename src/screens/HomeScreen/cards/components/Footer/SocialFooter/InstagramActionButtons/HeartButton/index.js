import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setActiveEventId } from '../../../../../../../../actions';

import HeartButton from './HeartButton';

const mapDispatchToProps = dispatch => bindActionCreators({
  setActiveEventId,
}, dispatch);

export default connect(undefined, mapDispatchToProps)(HeartButton);
