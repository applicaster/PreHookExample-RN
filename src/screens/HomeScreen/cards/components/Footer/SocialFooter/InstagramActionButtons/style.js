import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  facebookActionButtons: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});
