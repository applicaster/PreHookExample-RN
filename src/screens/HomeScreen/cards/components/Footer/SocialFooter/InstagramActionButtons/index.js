import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
 } from 'react-native';
import { styles } from './style';
import HeartButton from './HeartButton';
import CommentButton from './CommentButton';

class InstagramActionButtons extends Component {
  render() {
    const { commentsCount, likesCount, eventId } = this.props;
    return (
      <View style={[styles.facebookActionButtons]}>
        <HeartButton key={'likes'} eventId={eventId} likesCount={likesCount} />
        <CommentButton key={'comments'} eventId={eventId} commentsCount={commentsCount} />
      </View>
    );
  }
}

InstagramActionButtons.propTypes = {
  eventId: PropTypes.string.isRequired,
  commentsCount: PropTypes.number.isRequired,
  likesCount: PropTypes.number.isRequired,
};

export default InstagramActionButtons;
