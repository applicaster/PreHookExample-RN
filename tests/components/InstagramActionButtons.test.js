import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import InstagramActionButtons from '../../src/components/InstagramActionButtons';

jest.mock('../../src/components/CommentButton', () => 'CommentButton');
jest.mock('../../src/components/HeartButton', () => 'HeartButton');

describe('InstagramActionButtons', () => {
  test('component renders correctly', () => {
    const html = renderer.create(
      <InstagramActionButtons />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
