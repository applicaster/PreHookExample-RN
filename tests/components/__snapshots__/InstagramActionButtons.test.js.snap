// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstagramActionButtons component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <HeartButton
    eventId={undefined}
    likesCount={undefined}
  />
  <CommentButton
    commentsCount={undefined}
    eventId={undefined}
  />
</View>
`;
